server/sonar-web/src/main/js/apps/quality-profiles/changelog/ChangelogContainer.js

```diff
@@ -17,40 +17,52 @@
  * along with this program; if not, write to the Free Software Foundation,
  * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
  */
+// @flow
 import React from 'react';
 import Changelog from './Changelog';
 import ChangelogSearch from './ChangelogSearch';
 import ChangelogEmpty from './ChangelogEmpty';
 import { getProfileChangelog } from '../../../api/quality-profiles';
-import { ProfileType } from '../propTypes';
 import { translate } from '../../../helpers/l10n';
-
-export default class ChangelogContainer extends React.Component {
-  static propTypes = {
-    location: React.PropTypes.object.isRequired,
-    profile: ProfileType
-  };
+import { getProfileChangelogPath } from '../utils';
+import type { Profile } from '../propTypes';
+
+type Props = {
+  location: {
+    query: {
+      since?: string,
+      to?: string
+    }
+  },
+  organization: ?string,
+  profile: Profile
+};
+
+type State = {
+  events?: Array<*>,
+  loading: boolean,
+  page?: number,
+  total?: number
+};
+
+export default class ChangelogContainer extends React.PureComponent {
+  mounted: boolean;
+  props: Props;
 
   static contextTypes = {
     router: React.PropTypes.object
```

Comment:
Not really, this `contextTypes` definition is required to use context

see more: https://facebook.github.io/react/docs/context.html